client.go

```diff
@@ -0,0 +1,82 @@
+package symbiosis
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/go-resty/resty/v2"
+	"time"
+)
+
+const (
+	apiEndpoint = "https://api.symbiosis.host"
+)
+
+type Client struct {
+	symbiosisAPI *resty.Client
+}
+
+type ClientOption func(c *resty.Client)
+
+func WithTimeout(timeout time.Duration) ClientOption {
+	return func(c *resty.Client) {
+		c.SetTimeout(timeout)
+	}
+}
+
+func WithAlternativeEndpoint(endpoint string) ClientOption {
+	return func(c *resty.Client) {
+		c.SetHostURL(endpoint)
+	}
+}
+
+func NewClient(apiKey string, opts ...ClientOption) (*Client, error) {
```

Comment:
I'm no go-expert but I'm thinking, as we will probably have to add more config options later on would it make more sense for it to accept a struct with settings?
```go
type Client struct {
  baseURL string
  timeout time.Duration
  ..more to come..
}
```

That way we don't have to expose the resty client to users, however it is a bit less flexible.

Reply:
I did give this a fair bit of thought, I wanted to make all config options flexible so I opted for this approach:


```go
// Changing the symbiosis API endpoint
client, err := symbiosis.NewClient(os.Getenv("SYMBIOSIS_API_KEY"), symbiosis.WithAlternativeEndpoint("https://some-other-url"))

// Setting a default timeout
client, err := symbiosis.NewClient(os.Getenv("SYMBIOSIS_API_KEY"), symbiosis.WithTimeout(time.Second * 30)))
```

I don't see the need for a huge amount of configurability there. In theory we can have as many ClientOption functions as we'd need.

Reply:
That makes sense. Should we change `WithAlternativeEndpoint` to `WithEndpoint`? As many clients will always supply it from env variables and don't know if it's alternative or not.

One option could also be something like this:

```go
func (c *Client) WithEndpoint(url string) {
  ...
}

client, err := symbiosis.NewClient(os.Getenv("SYMBIOSIS_API_KEY")).
  WithEndpoint("staging.api.symbiosis.host")
```